agent/src/test/java/com/appland/appmap/record/RecorderTest.java

```diff
@@ -98,6 +104,78 @@ public void testUnwriteableOutputFile() throws IOException {
     assertTrue(exception.toString().indexOf("java.lang.RuntimeException: ") == 0);
   }
 
+  @Test
+  // for Files.move when: REPLACE_EXISTING, ATOMIC_MOVE
+  public void testWriteFileAcrossFilesystems() throws IOException {
+    if (!System.getProperty("os.name").equals("Linux")) {
+        //  /dev/shm exists only on Linux
+        return;
+    }
+    String sourceFilename = "/tmp/recordertest_file";
+    String targetFilename = "/dev/shm/recordertest_file";
+    File sourceFile = new File(sourceFilename);
+    File targetFile = new File(targetFilename);
+    Exception exception = null;
+    try {
+      // if the file exists createNewFile returns false
+      sourceFile.createNewFile();
+    } catch (IOException e) {
+    }
+
+    // Copying a file across filesystems should not throw an exception.
+    final Recording recording = new Recording("recorderName", sourceFile);
+    try {
+        recording.moveTo(targetFilename);
+    } catch (RuntimeException e) {
+        exception = e;
+    }
+
+    assertNull(exception);
+
+    sourceFile.delete();
+    targetFile.delete();
+  }
```

Comment:
```suggestion
    try {
        recording.moveTo(targetFilename);
    } catch (RuntimeException e) {
        exception = e;
    } finally {
        sourceFile.delete();
        targetFile.delete();
    }
    assertNull("recording.moveTo failed, " + exception.getMessage(), exception);
  }
```

Putting the calls to `delete` into a finally block will ensure that they're executed even if `moveTo` fails. 

Also, it's helpful to put a message in a generic assertion like `assertNull`. Otherwise, all the error message says is "expected null".

Reply:
Ok. Since the test isn't expecting to see a `RuntimeException` here, please be sure to print out the exception's message in the `catch` block.

Reply:
Added.